### PR TITLE
Add AWS CI to telemetry CI checks

### PR DESCRIPTION
**Pull request by ancalita**
**Proposed changes**:
- As a result of findings for issue #8694, I added [AWS CI environment variables](https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-env-vars.html) to the `telemetry` module to improve tracking of all CI pipelines.

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [X] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/telemetry.py [hunk 1]
```diff
@@ -72,6 +72,9 @@
     "JENKINS_URL",
     "TEAMCITY_VERSION",
     "TRAVIS",
+    "CODEBUILD_BUILD_ARN",
+    "CODEBUILD_BUILD_ID",
+    "CODEBUILD_BATCH_BUILD_IDENTIFIER",
```
Comment (wochinge) [inline rasa/telemetry.py:77]: why do we add all three? Would one be enough?

**Review by wochinge — APPROVED** [review]
🚀

**Comment by wochinge** [comment]
Is there actually also one for gitlab?

File: rasa/telemetry.py (hunk 1 above)
Comment (ancalita) [reply rasa/telemetry.py:77]: I wasn't sure which one is always present 😅 the language in the documentation wasn't very specific.

**Comment by wochinge** [comment]
ah, that's covered by `CI` https://docs.gitlab.com/ee/ci/variables/predefined_variables.html 👍🏻

File: rasa/telemetry.py (hunk 1 above)
Comment (wochinge) [reply rasa/telemetry.py:77]: at least `CODEBUILD_BUILD_ID` and `CODEBUILD_BUILD_ARN` seem to overlap from the documentation, no?

Comment (ancalita) [reply rasa/telemetry.py:77]: Oh yes, the `ARN` contains the `ID` 🙈 I've already merged `main` into the branch, would it be ok to keep this as it is?

Comment (wochinge) [reply rasa/telemetry.py:77]: sure 👍🏻

**Pull request merged**